typespec-tests/src/main/java/com/cadl/patch/models/Resource.java

```diff
@@ -425,55 +433,37 @@ private JsonWriter toJsonMergePatch(JsonWriter jsonWriter) throws IOException {
     @Generated
     public static Resource fromJson(JsonReader jsonReader) throws IOException {
         return jsonReader.readObject(reader -> {
-            String id = null;
-            String name = null;
-            String description = null;
-            Map<String, InnerModel> map = null;
-            Long longValue = null;
-            Integer intValue = null;
-            ResourceEnumValue enumValue = null;
-            InnerModel innerModelProperty = null;
-            List<InnerModel> array = null;
-            Fish fish = null;
+            Resource deserializedResource = new Resource();
             while (reader.nextToken() != JsonToken.END_OBJECT) {
                 String fieldName = reader.getFieldName();
                 reader.nextToken();
 
                 if ("id".equals(fieldName)) {
-                    id = reader.getString();
+                    deserializedResource.id = reader.getString();
                 } else if ("name".equals(fieldName)) {
-                    name = reader.getString();
+                    deserializedResource.name = reader.getString();
                 } else if ("description".equals(fieldName)) {
-                    description = reader.getString();
+                    deserializedResource.description = reader.getString();
                 } else if ("map".equals(fieldName)) {
-                    map = reader.readMap(reader1 -> InnerModel.fromJson(reader1));
+                    Map<String, InnerModel> map = reader.readMap(reader1 -> InnerModel.fromJson(reader1));
+                    deserializedResource.map = map;
                 } else if ("longValue".equals(fieldName)) {
-                    longValue = reader.getNullable(JsonReader::getLong);
+                    deserializedResource.longValue = reader.getNullable(JsonReader::getLong);
                 } else if ("intValue".equals(fieldName)) {
-                    intValue = reader.getNullable(JsonReader::getInt);
+                    deserializedResource.intValue = reader.getNullable(JsonReader::getInt);
```

Comment:
I’m not sure if we want to use the accessor pattern in ‘fromJson’ both for polymorphic json merge patch models and non-polymorphic json merge patch models. I tend to use accessor pattern for both, but not strong opinion.

Reply:
For now, I'd say let's go with using the access pattern for polymorphic properties only in JSON merge patch models for now and if needed we can use it everywhere. There's an additional change we'd want to make before enabling this everywhere which is determining if all the generated models are in the same package, if so package-private can be used, otherwise access helpers are needed. I didn't do that for JSON merge patch as:

1. It already was using access helpers to enable JSON merge patch serialization.
2. There aren't many cases of polymorphism + JSON merge patch, so the code noise is limited in scope.

Non-JSON merge patch polymorphism is much more common and I want to hold off on that larger change for a bit. And the non-JSON merge patch polymorphism can use the public setters of the parent models (assuming non-readonly, which won't work at this time anyway) to set the value.

Reply:
I'm fine with it.